pkg/fleet-manager/fleet_plugin_metric.go

```diff
@@ -0,0 +1,335 @@
+				OwnerReferences: []metav1.OwnerReference{
+					ownerReference,
+				},
+			},
+			Spec: corev1.ServiceSpec{
+				Ports: []corev1.ServicePort{
+					{
+						Name:        "grpc",
+						Port:        10901,
+						Protocol:    corev1.ProtocolTCP,
+						AppProtocol: pointer.String("grpc"),
+					},
+				},
+				ClusterIP: NoneClusterIP,
+			},
+		}
+
+		if err := f.Client.Create(context.Background(), svc); err != nil {
+			return err
+		}
+	}
+
+	if !capiutil.HasOwnerRef(svc.OwnerReferences, ownerReference) {
+		svc.OwnerReferences = append(svc.OwnerReferences, ownerReference)
+		if err := f.Client.Update(context.Background(), svc); err != nil {
+			return err
+		}
+	}
+
+	ep := &corev1.Endpoints{}
+	if err := f.Client.Get(context.Background(), svcNN, ep); err != nil {
+		if !apierrors.IsNotFound(err) {
+			return err
+		}
+
+		ep = &corev1.Endpoints{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      svcNN.Name,
+				Namespace: svcNN.Namespace,
+			},
+			Subsets: convertToSubset(endpoints),
+		}
+
+		if err := f.Client.Create(context.Background(), ep); err != nil {
+			return err
+		}
+	}
+
+	subsets := convertToSubset(endpoints)
+	if reflect.DeepEqual(ep.Subsets, subsets) {
+		return nil
+	}
+
+	ep.Subsets = subsets
+	if err := f.Client.Update(context.Background(), ep); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// syncObjstoreSecret syncs the secret to the cluster
+func (f *FleetManager) syncObjstoreSecret(ctx context.Context, fleetCluster *fleetCluster, secret *corev1.Secret) error {
+	_, err := fleetCluster.client.KubeClient().CoreV1().Namespaces().Get(ctx, secret.Namespace, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		_, err := fleetCluster.client.KubeClient().CoreV1().Namespaces().Create(ctx, &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: secret.Namespace,
+			},
+		}, metav1.CreateOptions{})
+		if err != nil {
+			return nil
+		}
+	} else if err != nil {
+		return nil
+	}
+
+	s, err := fleetCluster.client.KubeClient().CoreV1().Secrets(secret.Namespace).Get(ctx, secret.Name, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		_, err := fleetCluster.client.KubeClient().CoreV1().Secrets(secret.Namespace).Create(ctx, secret, metav1.CreateOptions{})
+		if err != nil {
+			return err
+		}
+	} else if err != nil {
+		return err
+	}
+
+	s.Data = secret.Data
+	_, err = fleetCluster.client.KubeClient().CoreV1().Secrets(secret.Namespace).Update(ctx, s, metav1.UpdateOptions{})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (f *FleetManager) reconcileMetricPlugin(ctx context.Context, fleet *fleetapi.Fleet, fleetClusters map[string]*fleetCluster) (kube.ResourceList, ctrl.Result, error) {
+	log := ctrl.LoggerFrom(ctx).WithValues("fleet", client.ObjectKeyFromObject(fleet))
+
+	if fleet.Spec.Plugin == nil ||
+		fleet.Spec.Plugin.Metric == nil {
+		// reconcilePluginResources will delete all resources if plugin is nil
+		return nil, ctrl.Result{}, nil
+	}
+
+	var (
```

Comment:
It is not recommended to declare var in go

Reply:
why?

Reply:
I remember once i used var, the lint in istio report warning

Reply:
did not see lint error in golang for following code:

https://github.com/istio/istio/blob/047d8e3558d0554a5e91d802ceec74573b55b10f/pkg/kube/inject/webhook_test.go#L890

Reply:
:= is comonly used for declare and assign, please keep consistent